advanced_functionality/multi_model_linear_learner_home_value/sklearn_preprocessor.py

```diff
@@ -0,0 +1,214 @@
+from __future__ import print_function
+
+import time
+import sys
+from io import StringIO
+import os
+import shutil
+
+import argparse
+import csv
+import json
+import numpy as np
+import pandas as pd
+
+from sklearn.compose import ColumnTransformer
+from sklearn.externals import joblib
+from sklearn.impute import SimpleImputer
+from sklearn.pipeline import Pipeline
+from sklearn.preprocessing import Binarizer, StandardScaler, OneHotEncoder
+
+from sagemaker_containers.beta.framework import (
+    content_types, encoders, env, modules, transformer, worker)
+
+# Since we get a headerless CSV file we specify the column names here.
+feature_columns_names = [
+    'YEAR_BUILT',  
+    'SQUARE_FEET',  
+    'NUM_BEDROOMS',  
+    'NUM_BATHROOMS', 
+    'LOT_ACRES',
+    'GARAGE_SPACES', 
+    'FRONT_PORCH', 
+    'DECK']
+
+label_column = 'PRICE'
+
+feature_columns_dtype = {
+    'YEAR_BUILT': str,
+    'SQUARE_FEET': np.float64,
+    'NUM_BEDROOMS': np.float64,
+    'NUM_BATHROOMS': np.float64,
+    'LOT_ACRES': np.float64,
+    'GARAGE_SPACES': np.float64,
+    'FRONT_PORCH': str,
+    'DECK': str}
+
+label_column_dtype = {'PRICE': np.float64}  
+
+
+def merge_two_dicts(x, y):
+    z = x.copy()  # start with x's keys and values
+    z.update(y)  # modifies z with y's keys and values & returns None
+    return z
+
+
+if __name__ == '__main__':
+
+    parser = argparse.ArgumentParser()
+
+    # Sagemaker specific arguments. Defaults are set in the environment variables.
+    parser.add_argument('--output-data-dir', type=str, default=os.environ['SM_OUTPUT_DATA_DIR'])
+    parser.add_argument('--model-dir', type=str, default=os.environ['SM_MODEL_DIR'])
+    parser.add_argument('--train', type=str, default=os.environ['SM_CHANNEL_TRAIN'])
+
+    args = parser.parse_args()
+
+    # Take the set of files and read them all into a single pandas dataframe
+    input_files = [os.path.join(args.train, file) for file in os.listdir(args.train)]
+    if len(input_files) == 0:
+        raise ValueError(('There are no files in {}.\n' +
+                          'This usually indicates that the channel ({}) was incorrectly specified,\n' +
+                          'the data specification in S3 was incorrectly specified or the role specified\n' +
+                          'does not have permission to access the data.').format(args.train, "train"))
```

Comment:
`"train"` can be placed directly in the body of the template, no formatting is needed.